pebble.go

```diff
@@ -363,6 +363,14 @@ func (b *pebbleDBBatch) Close() error {
 	return nil
 }
 
+// GetByteSize implements Batch
+func (b *pebbleDBBatch) GetByteSize() (uint64, error) {
+	if b.batch == nil {
+		return 0, errBatchClosed
+	}
+	return uint64(b.batch.Len()), nil
```

Comment:
Just commenting a thought here. Was slightly concerned about overflows on 32-bit machines but it seems that the max supported batch size would be 4Gb.

The batch should never be that large. So I think it's not a point of concern here

Reply:
I'll switch to uint32

Reply:
yeah, int is roughly 2GB

Reply:
I agree. Let's do `int` for now because we use `len()` in many places that would have returned an `int` anyways.

If we go with `int`, we should check for overflow in pebble's implementation because it has a native method returning `unit32`: https://github.com/cosmos/cosmos-db/blob/0727d99004160ab26308819ec76b59077f0bf86e/pebble.go#L371

Something like:

```go
if b.batch.Len() > int32MaxValue {
      return errors.New(...)
}
```

Another concern I have is that some implementations such as rocks include some internal metadata in their batch calculations. As a result, the max supported value of actual data size on 32-bit machines is less than 2GB. 

I think we don't need to worry about this problem at the moment. The batch should never be reaching this high max value so this is more of trying to be considerate of unlikely edge cases. However, we should keep it in mind when starting to benchmark the final result of this feature. To prevent these unlikely overflows,  we might want to add certain safeguards in IAVL against allowing the batch to grow past a certain threshold

Reply:
yeah I agree with you that we should not be so concerned about that unlikely overflow edge cases cuz we try to prevent that edge cases anyway

Reply:
> If we go with `int`, we should check for overflow in pebble's implementation because it has a native method returning `unit32`:

why do we need to check for overflow ? there won't be overflow since won't use typecast anyway, we just need to remove `uint32` type cast in `return uint32(b.batch.Len()), nil ` and `b.batch.Len()` will do it job returning an `int`.